src/codegen/build_module.cc

```diff
@@ -392,7 +392,7 @@ Stmt BuildStmt(Schedule sch,
 
   // Phase 1
   stmt = ir::StorageFlatten(stmt, out_binds, 64,
-                            config->instrument_bound_checkers);
+                            config->instrument_bound_checkers, bounds);
```

Comment:
We don't have to get bounds from the previous scheduling stage, instead, we could get them from the range of the loops an bind these information with analyzer->Bind related https://github.com/dmlc/tvm/blob/master/src/arithmetic/ir_mutator_with_analyzer.h

Reply:
@tqchen, ir mutator would imply you want to mutate exprs whereas all we want is the bound information. Can we not just get new bounds after ScheduleOps?

Reply:
So that we have bounds information from the new schedule.

Reply:
Oh, i do not mean we should reuse the IR mutator, but simply means we could populate the bounds in the same way as the IRMutatorWithAnalyzer in the visitor.

The main reason to re-populate these bounds is to make sure that StorageFlatten stand on its own as a pass. As these information are already available in the context.  So that we can apply the same pass for code that may not have bound information available

Reply:
@tqchen, to make sure, you are suggesting that we populate bounds inside StorageFlatten and now pass it from outside? Sorry to drag this, but just want to make sure.

Reply:
Yes, hat i mean is to populate the bounds(from the loops) inside the StorageFlatten pass, so the pass itself can be self-contained without relying out the external information :) . Thanks for the clarification questions. It is great that we can do these kind of discussions in public, as the apache way always encourages public discussions and allow other developers to follow the development.

Reply:
@tqchen, updated the PR. Hopefully this is akin to what you had in mind :).